src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext';
+import * as PostService from '../../api/todos';
+import { ErrorMessage } from '../../types/ErrorMessageEnum';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+}) => {
+  const {
+    todos,
+    setTodos,
+    handleDelete,
+    changingId,
+    setErrorOccured,
+  } = useContext(TodoContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [dbClicked, setDbClicked] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [isXButtonVisible, setisXButtonVisible] = useState(true);
+
+  const editingRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editingRef.current && dbClicked) {
+      editingRef.current.focus();
+    }
+  }, [dbClicked]);
+
+  useEffect(() => {
+    setIsLoading(changingId.includes(todo.id));
+  }, [changingId]);
```

Comment:
<img width="663" alt="image" src="https://github.com/mate-academy/react_todo-app-with-api/assets/70220667/082d40ac-6ec3-4a16-ba72-f05c203da7dc">
better to fix it